api/datadoghq/v1alpha2/datadogpodautoscaler_conversion_test.go

```diff
@@ -0,0 +1,297 @@
+package v1alpha2
+
+import (
+	"testing"
+
+	"github.com/DataDog/datadog-operator/api/datadoghq/common"
+	"github.com/DataDog/datadog-operator/api/datadoghq/v1alpha1"
+	"github.com/DataDog/datadog-operator/api/utils"
+	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/assert"
+	v2 "k8s.io/api/autoscaling/v2"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestNewDatadogPodAutoscalerFromV1Alpha1(t *testing.T) {
+	in := &v1alpha1.DatadogPodAutoscaler{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "DatadogPodAutoscaler",
+			APIVersion: "datadoghq.com/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-dpa",
+			Namespace: "default",
+		},
+		Spec: v1alpha1.DatadogPodAutoscalerSpec{
+			TargetRef: v2.CrossVersionObjectReference{
+				Kind:       "Deployment",
+				Name:       "test-deployment",
+				APIVersion: "apps/v1",
+			},
+			Owner:         "test-owner",
+			RemoteVersion: utils.NewPointer[uint64](10),
```

Comment:
```suggestion
			RemoteVersion: utils.NewUInt64Pointer(10),
```
See comment on `utils.go` file